test/benchmark/benchmark_script.py

```diff
@@ -0,0 +1,528 @@
+import argparse
+import re
+import shutil
+import os
+import pickle
+import subprocess
+
+from matplotlib import pyplot as plt
+import seaborn as sns
+
+import nmodl.dsl as nmodl
+
+class CompilersConfig:
+    svml_lib = ""
+    intel_exe = ""
+    sleef_lib = ""
+    clang_exe = ""
+    llc_exe = ""
+    gcc_exe = ""
+    libdevice_lib = ""
+    nmodl_exe = ""
+
+    def __init__(
+        self,
+        svml_lib,
+        intel_exe,
+        sleef_lib,
+        clang_exe,
+        llc_exe,
+        gcc_exe,
+        libdevice_lib,
+        nmodl_exe,
+    ):
+        self.svml_lib = svml_lib
+        self.intel_exe = intel_exe
+        self.sleef_lib = sleef_lib
+        self.clang_exe = clang_exe
+        self.llc_exe = llc_exe
+        self.gcc_exe = gcc_exe
+        self.libdevice_lib = libdevice_lib
+        self.nmodl_exe = nmodl_exe
+
+
+class BenchmarkConfig:
+    mod_files = ""
+    architectures = ""
+    compilers = ""
+    math_libraries = ["SVML", "SLEEF"]
+    fast_math = [False, True]
```

Comment:
I see how you use this below but i find it a bit weird - maybe I'll propose a better way in a separate commit

Reply:
If you have a better suggestion feel free to add it :+1: We can also discuss about that if you want